napari/layers/shapes/_shape_list.py

```diff
@@ -85,7 +85,7 @@ class ShapeList:
     """
 
     def __init__(
-        self, data: typing.Iterable[Shape] = (), ndisplay: int = 2
```

Comment:
Shouldn't this (and all the other related lines) be `Literal[2, 3]`?

Reply:
For shapes `ndisplay` is always 2 according to https://github.com/napari/napari/blob/340f330eb473a711eaf12fdbbf7d3e72074d72a5/napari/layers/shapes/_shapes_models/shape.py#L51-L53, and the typing here follows from that.

Reply:
That seems off, unless I'm misunderstanding. 
You can't edit Shapes layers in 3D, but shapes are displayed in 3D...